liminal/actions/AssistantMessage.ts

```diff
@@ -16,7 +17,7 @@ export function* assistant(
   }>,
   void
 > {
-  const content = isTemplateStringsArray(raw) ? String.raw(raw, ...substitutions) : raw
+  const content = isTemplateStringsArray(raw) ? dedent(String.raw(raw, ...substitutions)) : raw
```

Comment:
Change: You need to dedent before substitutions, or else you can screw up the indentation in the substitutions (I have a sample here: https://gist.github.com/colelawrence/52cd872480e160447eb3ec68bc0d4ff0)